frontend/source/components/AvailabilityComponent/AvailabilityComponent.js

```diff
@@ -0,0 +1,107 @@
+import { EventHub } from "../../eventhub/EventHub.js";
+import { Events } from "../../eventhub/Events.js";
+import { BaseComponent } from "../BaseComponent/BaseComponent.js";
+
+export class AvailabilityComponent extends BaseComponent {
+    #container = null;
+    #potentialTimes = []; //{startTime, endTime, date} // need to get from indexedDB
+    #eventName = "{name}"; // need to get from indexedDB
+    #eventDescription = "{description}"; // need to get from indexedDB
+
+    constructor() {
+        super();
+        this.loadCSS("AvailabilityComponent");
+    }
+
+    render() {
+        if(this.#container){
+            return this.#container;
+        }
+
+        this.#createContainer();
+        this.#attachEventListeners();
+        (async () => {let permission = await Notification.requestPermission();})();
+        return this.#container;
+    }
+
+    #createContainer() {
+        // Create and configure the container element
+        this.#container = document.createElement("div");
+        this.#container.classList.add("availability-input-container");
+        this.#getEventDetails();
+        this.#container.innerHTML = this.#getTemplate();
+        this.#addTimes();
+    }
+
+    #getEventDetails() {
+        // get event details from db
+        // test data:
+        this.#eventName = "sample name";
+        this.#eventDescription = "sample sample sample sample sample"
+        this.#potentialTimes.push({startTime: "01:00", endTime: "02:00", date: "1970-01-01"});
+        this.#potentialTimes.push({startTime: "02:00", endTime: "03:00", date: "1970-01-01"});
+    }
+
+    #getTemplate() {
+        // Returns the HTML template for the component
+        return `
+            <div id="blue-header">
+                <h1>Submit Available Times</h1>
+            </div>
+
+            <div id="availability-input-body">
+                <h2>Event Details</h2>
+                <h3 id="eventName">${this.#eventName}</h3>
+
+                <h3 id="eventDescription">${this.#eventDescription}</h3>

```

Comment:
I dont think the `${}` format works in HTML. Might have to reimplement this part with JS.